store/todos.js

```
@@ -61,42 +65,71 @@ export const getters = {
 export const actions = {
   // AddTask Operation
 
-  addTask: ({ state, commit }, val) => {
-    commit('addTask', val);
-    // set filter option to All
-    commit('filterTaskList');
+  addTask: async ({ state, commit }, val) => {
+    try {
+      const task = {
+        status: false,
+        description: val,
+        completed_at: null,
+        created_at: new Date(),
+      };
+      const { error } = await supabase.from('Todos').insert(task).single();
+
+      if (error) {
+        return { success: false };
+      }
+
+      state.filterOptions.forEach((element) => (element.status = false));
+      state.filterOptions[0].status = true;
+      state.activeFilterOption = state.filterOptions[0];
+
+      commit('filterTaskList');
+
+      return { success: true };
+    } catch (err) {}
```

Can we throw an error here?

---

resolved